internal/core/domain/market_service.go

```diff
@@ -217,6 +219,176 @@ func (m *Market) ChangeQuotePrice(price decimal.Decimal) error {
 	return nil
 }
 
+func (m *Market) Preview(
+	baseBalance, quoteBalance, amount uint64,
+	isBaseAsset, isBuy bool,
+) (*PreviewInfo, error) {
+	if !m.IsTradable() {
+		return nil, ErrMarketIsClosed
+	}
+
+	if isBaseAsset {
+		if amount < uint64(m.FixedFee.BaseFee) {
```

Comment:
I would argue that also equal to fixed fee, is too low.

Also we do not take into consideration that if percentage fee is greater than zero, wuold not be enough.

Reply:
> Also we do not take into consideration that if percentage fee is greater than zero, wuold not be enough.

The strategy takes care of charging the percentage fee, and, in case the fee must be subtracted, we check that after appliying the fees the amount is still positive.

It's not possible to say if the amount is "enough" for perc fees but doing the math "empirically" and checking the calculated amount is > 0.

> I would argue that also equal to fixed fee, is too low.

My though was: if the amount is at least equal to the respective fixed fee then it's allright because the perc fee will be applied. I'll add the equality check if this way of thinking does not convince you.

Reply:
receiving 0 output, although doable, not that much meaningful for the user